modules/n1ql/pages/n1ql-language-reference/groupby.adoc

```diff
@@ -49,6 +50,51 @@ For details, see xref:n1ql-language-reference/from.adoc#section_ax5_2nx_1db[AS C
 Assigning an alias to the group term is optional.
 If you assign an alias, the `AS` keyword may be omitted.
 
+[[group-as]]
+=== GROUP AS Clause
+
+[source,ebnf]
+----
+include::partial$grammar/dql.ebnf[tag=group-as-clause]
+----
+
+image::n1ql-language-reference/group-as-clause.png["Syntax diagram", align=left]
+
+JSON is a hierarchical format, and a fully featured JSON query language needs to be able to produce hierarchies of its own, with computed data at every level of the hierarchy. 
+The key feature of SQL{pp} that makes this possible is the `GROUP AS` clause.
+
+A query can include a `GROUP AS` clause only if it has a `GROUP BY` clause. 
+The effect of a `GROUP BY` clause is to hide the original objects in each group, exposing only the grouping expressions and special aggregation functions on the non-grouping fields. 
+The purpose of the `GROUP AS` clause is to make the original objects in the group visible to subsequent clauses. 
+As a result, the query can generate output data both for the group as a whole and for the individual objects inside the group.
+
+For each group, `GROUP AS` preserves all of the objects in the group, as they were before grouping, in an array. 
+The name of the array is the alias in the GROUP AS clause. 
+You can then use the array name in the `FROM` clause of a subquery to process and return the individual objects in the group.
+
+In the `GROUP AS` array, each object is wrapped in an outer object that gives it the name of the alias specified in the `FROM` clause, or its implicit alias if no alias was specified.
+Identifying the objects with their aliases in this way helps avoid ambiguity.
+
+For example, a query has a `FROM` clause of `FROM airline as a, route as r`, and a `GROUP AS` clause of `GROUP AS g`. 
+The `GROUP AS` clause preserves the airline and route objects together in the same array. 
```

Comment:
One minor change - could we please change "The `GROUP AS` clause preserves the airline and route objects together in the same array. " 

to

"Each item in the group would contain both an `airline` object and a `route` object."

[ the reason I ask for this change is because it is important for users to understand that each group _item_ will have both objects. And hence, requiring each object to be identified separately with its alias ]

Everything else looks great to me!

Reply:
got it. 
I am trying very hard not to use the word "group" to describe the JSON syntax that results from GROUP BY or GROUP AS since I think that overloads the word. 
How's this @dhanyagowrish : 

The result of the `GROUP AS` clause is an array of objects, each of which contains both an airline object and a route object.

Reply:
That sounds great.